src/field/pokemon.ts

```diff
@@ -3007,6 +3007,9 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
   damageAndUpdate(damage: integer, result?: DamageResult, critical: boolean = false, ignoreSegments: boolean = false, preventEndure: boolean = false, ignoreFaintPhase: boolean = false, source?: Pokemon): integer {
     const damagePhase = new DamageAnimPhase(this.scene, this.getBattlerIndex(), damage, result as DamageResult, critical);
     this.scene.unshiftPhase(damagePhase);
+    if (this.switchOutStatus && source && !source.hasAbilityWithAttr(PostDefendContactDamageAbAttr)) {
```

Comment:
```suggestion
    if (this.switchOutStatus && source) {
```

I don't think the `PostDefendContactDamageAbAttr` check is necessary right now, since the ability's code does not pass a `source` argument into `damageAndUpdate()`. Removing this check might be better to prevent potential edge cases, such as a Pokemon with Iron Barbs using a move to hit a Pokemon that has fled.
![image](https://github.com/user-attachments/assets/159692e2-f990-477e-8046-c72544b45e5c)